homeassistant/components/enasolar/sensor.py

```diff
@@ -0,0 +1,250 @@
+                or (
+                    not sensor.sensor.per_day_basis
+                    and not sensor.sensor.per_total_basis
+                )
+            ):
+                state_unknown = True
+            sensor.async_update_values(unknown_state=state_unknown)
+            _LOGGER.debug(
+                "Data Sensor %s updated => %s", sensor.sensor.key, sensor.native_value
+            )
+        return values
+
+    def start_update_interval(event):  # pylint: disable=unused-argument
+        """Start the update interval scheduling."""
+        config_entry.async_on_unload(
+            async_track_time_interval_backoff(
+                hass, async_enasolar_meters, SCAN_METERS_MIN_INTERVAL
+            )
+        )
+        config_entry.async_on_unload(
+            async_track_time_interval_backoff(
+                hass, async_enasolar_data, SCAN_DATA_MIN_INTERVAL
+            )
+        )
+
+    if hass.is_running:
+        start_update_interval(None)
+    else:
+        hass.bus.async_listen_once(EVENT_HOMEASSISTANT_START, start_update_interval)
+
+
+@callback
+def async_track_time_interval_backoff(hass, action, min_interval) -> CALLBACK_TYPE:
+    """Add a listener that fires repetitively and increases the interval when failed."""
+    remove = None
+    interval = min_interval
+
+    async def interval_listener(now=None):  # pylint: disable=unused-argument
+        """Handle elapsed interval with backoff."""
+        nonlocal interval, remove
+
+        try:
+            if await action():
+                interval = min_interval
+            else:
+                interval = min(interval * 2, SCAN_MAX_INTERVAL)
+        finally:
+            remove = async_call_later(hass, interval, interval_listener)
+
+    hass.async_create_task(interval_listener())
+
+    def remove_listener():
+        """Remove interval listener."""
+        remove()  # pylint: disable=not-callable
+
+    return remove_listener
+
+
+class EnaSolarSensor(SensorEntity):
+    """Representation of a EnaSolar sensor."""
+
+    def __init__(self, pyenasolar_sensor, inverter_name=None, serial_no=None):
+        """Initialize the EnaSolar sensor."""
+        self.sensor = pyenasolar_sensor
+        if inverter_name:
+            self._attr_name = f"{inverter_name}_{self.sensor.name}"
+        else:
+            self._attr_name = f"enasolar_{self.sensor.name}"
+        self.serial_no = serial_no
+        self._native_value = self.sensor.value
+
+        if pyenasolar_sensor.is_meter:
+            self._attr_state_class = STATE_CLASS_MEASUREMENT
+        else:
+            self._attr_state_class = STATE_CLASS_TOTAL_INCREASING
+        self._attr_native_unit_of_measurement = ENASOLAR_UNIT_MAPPINGS[self.sensor.unit]
+        self._attr_should_poll = False
+
+    @property
+    def name(self) -> str | None:
+        """Return the name of the sensor."""
+        if hasattr(self, "_attr_name"):
+            return self._attr_name
+        return None
+
```

Comment:
```suggestion
```

Reply:
So the property name is not needed?  I have actioned.